crud-operations/src/test/java/org/example/crudoperations/integratedtests/ProductIntegrationTest.java

```diff
@@ -0,0 +1,89 @@
+package org.example.crudoperations.integratedtests;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.example.crudoperations.CrudOperationsApplication;
+import org.example.crudoperations.entity.Product;
+import org.example.crudoperations.repository.ProductRepository;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.boot.test.web.client.TestRestTemplate;
+import org.springframework.boot.web.server.LocalServerPort;
+import org.springframework.http.*;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import java.util.Optional;
+import java.util.UUID;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = CrudOperationsApplication.class, webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class ProductIntegrationTest {
+
+    @LocalServerPort
+    private int port;
+
+    TestRestTemplate restTemplate=new TestRestTemplate();
+
+    HttpHeaders headers=new HttpHeaders();
+    @Autowired
+    ObjectMapper objectMapper;
+
+    @MockBean
```

Comment:
How is this different from the unit test if we are mocking the repository bean. Use h2 database for integration tests to actually test the database connection for it to be an integrated test.